tests/generic_config_updater/test_dynamic_acl.py

```diff
@@ -0,0 +1,778 @@
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+def dynamic_acl_remove_nonexistant_table(duthost):
+    """Remove a nonexistent ACL Table from the duthost, verify it fails"""
+    json_patch = [
+        {
+            "op": "remove",
+            "path": "/ACL_TABLE/DYNAMIC_ACL_TABLE_BAD",
+            "value": { }
+        }
+    ]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_failure(output)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+def dynamic_acl_remove_table_type(duthost):
+    """Remove an ACL Table definition from the duthost
+    As we only have one ACL Table definition on """
+    json_patch = [
+        {
+            "op": "remove",
+            "path": "/ACL_TABLE_TYPE",
+            "value": { }
+        }
+    ]
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+
+
+
+def test_gcu_acl_drop_rule_creation(rand_selected_dut, ptfadapter, setup, dynamic_acl_create_table):
+    dynamic_acl_create_drop_rule_initial(rand_selected_dut, setup)
+    dynamic_acl_verify_packets(setup, ptfadapter, packets = generate_drop_packets(setup), packets_dropped = True)
+    dynamic_acl_verify_packets(setup,
+                               ptfadapter,
+                               packets = generate_drop_packets(setup),
+                               packets_dropped = False,
+                               src_port = setup["unblocked_src_port_indice"])
+
+
+def test_gcu_acl_drop_rule_removal(rand_selected_dut, ptfadapter, setup, dynamic_acl_create_table):
+    dynamic_acl_create_drop_rule_initial(rand_selected_dut, setup)
+    dynamic_acl_remove_drop_rule_initial(rand_selected_dut)
+    dynamic_acl_verify_packets(setup, ptfadapter, packets = generate_drop_packets(setup), packets_dropped = False)
+
+def test_gcu_acl_forward_rule_priority_respected(rand_selected_dut, ptfadapter, setup, dynamic_acl_create_table):
+    dynamic_acl_create_forward_rules(rand_selected_dut)
+    dynamic_acl_create_drop_rule(rand_selected_dut, setup)
+    dynamic_acl_verify_packets(setup, ptfadapter, packets = generate_forward_packets(setup), packets_dropped = False)
+    dynamic_acl_verify_packets(setup, ptfadapter, packets = generate_drop_packets(setup), packets_dropped = True)
+
+def test_gcu_acl_forward_rule_replacement(rand_selected_dut, ptfadapter, setup, dynamic_acl_create_table):
+    dynamic_acl_create_forward_rules(rand_selected_dut)
+    dynamic_acl_create_drop_rule(rand_selected_dut, setup)
```

Comment:
The forward rule and drop rule are added for each case, but when are the rules deleted?

Reply:
Rules should be deleted at the end of each test case at the end of the autouse fixture "setup_env" via config rollback to initial state of device, if i am not mistaken.  Let me know if I am incorrect on this.